### ISSUE TITLE
[Bug]: Restore from Visual Studio fails after package changes

### ISSUE THREAD

**Issue by BertanAygun** [NuGet/Home#11749]
### NuGet Product Used

Visual Studio Package Manager Console

### Product Version

VS 32413.50.main

### Worked before?

VS 17.1

### Impact

It's more difficult to complete my work

### Repro Steps & Context

This happens when I pull changes from man branch that includes package version changes or sometimes target framework changes. A generic repro is:

- Open a solution from VS, do restore/build.
- Close VS
- Pull changes to the solution (from command line) that changes package versions, target frameworks.
- Open VS and open the solution
- Run "restore packages" from solution nodes context menu

I get `NETSDK1005` errors with description like `Assets file  <json> doesn't have a target for 'netstandard2.0'. Ensure that restore has run and   that you have included 'netstandard2.0' in the TargetFrameworks for your   project.`

The only workaround is to run restore from command line and once it succeeds VS continues fine. This impacts build from VS as well.

**Comment by nkolev92** [comment]
Sounds similar to https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1461787.

I'll attempt a repro.

**Comment by nkolev92** [comment]
[ConsoleApp1.zip](https://github.com/NuGet/Home/files/8498908/ConsoleApp1.zip)

I attempted a repro, but no luck. 

Back when the similar issue was reported, there was an experiment run.

Can you collect some logs for us for the second run (the one that fails, so the one after the branch switch). 


Set CPS_DiagnosticRuntime=1 before starting VS and that'll open an output window called Project. All that content would be relevant for us.

It would be great to point us to a commit sha so that we can see whether there's any difference in nomination vs what's on disk.

**Comment by BertanAygun** [comment]
Unfortunately, I don't have a consistent repro either. It seems to depend on when/where I built from before merging to tip. Next time I have a repro I will record the commits and also collect the log.

**Comment by nkolev92** [comment]
Thank you reporting this issue. At this point we do not have enough information to continue investigating the issue. 
Please feel free to comment or open a new issue if needed.

@BertanAygun Feel free to ping us if you hit this again.
